eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,57 @@
+function Get-NamepspacesFromDll($dllPath) {
+    $file = [System.IO.File]::OpenRead($dllPath)
+    try {
+        # Use to parse the namespaces out from the dll file.
+        $pe = [System.Reflection.PortableExecutable.PEReader]::new($file)
+        try {
+            $meta = [System.Reflection.Metadata.PEReaderExtensions]::GetMetadataReader($pe)
+            foreach ($typeHandle in $meta.TypeDefinitions) {
+                $type = $meta.GetTypeDefinition($typeHandle)
+                $attr = $type.Attributes
+                if ($attr -band 'Public' -and !$type.IsNested) {
+                    [pscustomobject]@{
+                        Name = $meta.GetString($type.Name)
+                        Namespace = $meta.GetString($type.Namespace)
+                    }
+                }
+            }
+        } finally {
+            $pe.Dispose()
+        }
+    } finally {
+        $file.Dispose()
+    }
+}
+
+function Fetch-NamespacesFromNupkg ($nupkgFilePath, $destination) {
+    $tempLocation = (Join-Path ([System.IO.Path]::GetTempPath()) "dotnetNupkg")
+    if (Test-Path $tempLocation) {
+        Remove-Item $tempLocation/* -Recurse -Force 
+    }
+    else {
+        New-Item -ItemType Directory -Path $tempLocation -Force | Out-Null
+    }
+
+    Write-Host "Unzipping nupkg..."
+    Write-Host $nupkgFilePath
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($nupkgFilePath, $tempLocation)
+    # .NET core includes multiple target framework. We currently have the same namespaces for different framework. 
+    # Will use whatever the first dll file.
+    Write-Host "Searching ddl file..."
+    $firstDllFiles = Get-ChildItem "$tempLocation/lib" -Filter '*.dll' -Recurse
```

Comment:
@heaths do you think we should prefer things under netstandard2.0?

Reply:
Yes, though practically it shouldn't matter. I believe Azure.Core is the only one we multi-target (I still want to open up more, though, which I'm discussing with @jsquire and @pallavit) but even in all cases we do, it's only to avoid using "reflection-based proxies" for implementation like I do for Key Vault. Namespaces shouldn't change, though.

For the foreseeable future, we will *always* target at least "netstandard2.0" so that should always be a safe fallback. Combined with that and that namespaces shouldn't differ, I think hardcoding to "netstandard2.0" is a good idea.

Reply:
Monitor.Ingestion.Logs and Tables I believe are also multi-targeted. @christothes to keep me honest here.

Reply:
For context, I support moving towards wider multitargeting.   I agree with Heath that it would be a good idea to sniff the required target frameworks from each library.

Reply:
I didn't actually say that. :smile: We should *always* support netstandard2.0.

But it is possible in some future iteration we start supporting more advanced features that could be in subnamespaces. But rather than trying to plumb through the `<TargetFrameworks>` either by parsing it (which can inherit or even be overridden what you see in the .csproj), just get namespaces from all assemblies therein?

```pwsh
$namespaces = get-childitem "$tempLocation/lib" -filter *.dll -recurse | get-namespacesfromdll | sort-object -unique # or whatever you called the function
```

My [original function](https://gist.github.com/heaths/c15a0818bed06b68fb171252db3b1c64) I gave you took pipeline input for the file names per the path/literalpath convention; otherwise, you can just call it in a `foreach` loop if that's easier for you.

This would be far easier with no real downside I see.

Reply:
> I didn't actually say that. 😄 We should _always_ support netstandard2.0 so I was saying you should just always use that one.

My bad.   While I don't see us having an API shape that differs per-framework in the short-term, I'm also wary to rule out that we ever may do so.   I agree that using `netstandard2.0` as a fallback makes good sense - but I believe we'd have to sniff frameworks and parse builds independently if some namespaces are available only in a subset, no?

Reply:
> Monitor.Ingestion.Logs and Tables I believe are also multi-targeted. @christothes to keep me honest here.

Correct on Logs, but not Tables.

Reply:
Yes, I agree we should just enumerate all DLLs. If the target framework moniker (TFM) is important for grouping, the subdirectory name and TFM are the same so you can do something like this instead:
```pwsh
$namespaces = get-childitem "$tempLocation/lib" -filter *.dll -recurse | foreach-object {
  $tfm = $_ | split-path -parent | split-path -leaf
  $_ | get-namespacefromdll | sort-object namespace -unique | add-member -type noteproperty -name TFM -value $tfm -passthru
}
```

You'll get back a bunch of objects sorted by namespace with only unique namespaces and `Namespace` and `TFM` properties you can use to group or whatever. Again, no need to try to parse out `<TargetFrameworks>` from the associated project file (much harder to do properly).

Reply:
While I kind of like the idea of aggregating all the namespaces I don't think we need to do that at least right now as we only support ref docs for a single TFM and that is netstandard2.0 for all track 2 libraries. Which is why I think netstandard2.0 will probably be enough for our purposes here.

The only places where that might not work is for any track 1 libraries we are still supporting that don't have a netstandard2.0 target, not sure if there are any but that is my only concern so we might want to default to netstandard2.0 and if it doesn't exist then just pick the first one.

Reply:
Current proposal is to parse out `netstandard2.0` namespaces into {package}.json as a new property. Toc generation will read namespaces from {package.json}. 

If namespaces missing in package.json, we still have the ability to download the nuget package from public repository. We can read whatever the first target framework if netstandard2.0 is missing.